src/file/FileDialog.tsx

```diff
@@ -215,6 +218,74 @@ const OpenLocalFile: React.FC<SourceTabProps> = ({ onDismiss }) => {
     );
 };
 
+const ImportFromString: React.FC<SourceTabProps> = ({ onDismiss }) => {
+    const loadScene = useLoadScene();
+    const setSavedState = useSetSavedState();
+    const isDirty = useIsDirty();
+    const theme = useTheme();
+    const [data, setData] = useState<string | undefined>('');
+    const [error, setError] = useState<string | undefined>('');
+
+    const importCallback = useCallback(async () => {
+        if (!data) {
+            return;
+        }
+
+        if (isDirty) {
+            if (!(await confirmUnsavedChanges(theme))) {
+                return;
+            }
+        }
+
+        let scene;
+        try {
+            scene = await textToScene(data);
+        } catch (ex) {
+            console.log(ex);
+            setError('Invalid Import String');
+            return;
+        }
+
+        loadScene(scene, undefined);
+        setSavedState(scene);
+        onDismiss?.();
+    }, [data, isDirty, theme, loadScene, setSavedState, onDismiss]);
+
+    const onKeyChange = useCallback(
+        (ev: FormEvent<HTMLInputElement | HTMLTextAreaElement>, value?: string) => {
+            setData(value);
+            setError(undefined);
+        },
+        [setError, setData],
+    );
+
+    const onKeyPress = useCallback(
+        (ev: React.KeyboardEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+            if (ev.key === 'Enter') {
+                importCallback();
+            }
+        },
+        [importCallback],
+    );
+
+    return (
+        <>
+            <TextField
+                label="Enter Import String"
+                multiline
+                rows={7}
+                onChange={onKeyChange}
+                onKeyPress={onKeyPress}
```

Comment:
`onKeyPress` is apparently deprecated (I'm guessing this was added to the react types sometime after I initially wrote this). Does using `onKeyDown` or `onKeyUp` work here?

Reply:
Both work - `onKeyDown` appears to have the most intuitive behavior. Using that.